packages/fxa-content-server/tests/functional/settings_v2/lib/helpers.js

```diff
@@ -33,6 +38,62 @@ async function navigateToSettingsV2(remote) {
   return email;
 }
 
+async function generateSecurityCode(remote) {
+  await click(
+    selectors.SETTINGS_V2.SECURITY.TFA.ADD_BUTTON,
+    selectors.SETTINGS_V2.SECURITY.TFA.SECURITY_CODE_TEXTBOX_LABEL,
+    remote
+  );
+  await click(selectors.SETTINGS_V2.SECURITY.TFA.SHOW_CODE_LINK, remote);
+
+  // store the secret key to use later
+  await testElementExists(
+    selectors.SETTINGS_V2.SECURITY.TFA.SECRET_CODE_MODAL,
+    remote
+  );
+  secret = await remote
+    .findByCssSelector(selectors.SETTINGS_V2.SECURITY.TFA.SECRET_CODE_TEXT)
+    .getVisibleText();
+
+  // store the security code generated via the secret code
+  securityCode = generateTotpCode(secret);
+  return securityCode;
+}
+
+async function confirmRecoveryCode(remote) {
+  securityCode = await generateSecurityCode(remote);
+  await click(
+    selectors.SETTINGS_V2.SECURITY.TFA.SECURITY_CODE_TEXTBOX_LABEL,
+    remote
+  );
+  await type(
+    selectors.SETTINGS_V2.SECURITY.TFA.SECURITY_CODE_TEXTBOX_INPUT,
+    securityCode,
+    remote
+  );
+  await click(selectors.SETTINGS_V2.SECURITY.TFA.CONTINUE_BUTTON, remote);
+  await testElementExists(
+    selectors.SETTINGS_V2.SECURITY.TFA.RECOVERY_CODE_BLOCK,
+    remote
+  );
+  const recoveryKey = await remote
+    .findByCssSelector(selectors.SETTINGS_V2.SECURITY.TFA.FIRST_RECOVERY_CODE)
+    .getVisibleText();
+  await click(selectors.SETTINGS_V2.SECURITY.TFA.CONTINUE_RECOVERY_KEY, remote);
+  await testElementExists(
+    selectors.SETTINGS_V2.SECURITY.TFA.RECOVERY_KEY_INPUT_LABEL,
+    remote
+  );
+  await type(
+    selectors.SETTINGS_V2.SECURITY.TFA.RECOVERY_KEY_INPUT,
+    recoveryKey,
+    remote
+  );
+  await click(selectors.SETTINGS_V2.SECURITY.TFA.SUBMIT_RECOVERY_KEY, remote);
+}
+
```

Comment:
I think these two new functions shouldn't be in the helpers as they are very 2FA specific.

Reply:
Yeah but then if I am writing separate tests for enable and disable, I will have to duplicate the same steps to first enable and then disable. Or if I can just write one whole test that first enables and then disables this could be removed. But I thought breaking functionalities into different tests is better. What do you suggest?

Reply:
Well, you can still keep them as functions, but in the test suite, not here.

Reply:
Oh, yes, it's fine to  have just one test for all the steps as well.